benchmarks/driver.py

```diff
@@ -81,8 +81,10 @@ def __init__(self, args, loglevel):
             for v in range(args.verbose - args.quiet):
                 self.extra_args.append("-v")
         # Set environment variables for dynamic loading (Linux and Mac)
-        os.putenv("LD_LIBRARY_PATH", f"{self.lib_dir}:{os.getenv('LD_LIBRARY_PATH')}")
-        os.putenv("DYLD_LIBRARY_PATH", f"{self.lib_dir}:{os.getenv('DYLD_LIBRARY_PATH')}")
+        for path in ["LD_LIBRARY_PATH", "DYLD_LIBRARY_PATH"]:
+            environ = [os.getenv(path)] if os.getenv(path) else []
+            environ.insert(0, self.lib_dir)  # prepend
+            os.environ[path] = ":".join(environ)
```

Comment:
Why is this better?
Just a general question as I'm not that familiar with python's environ

Reply:
Ahh ok it's just generalized the prepending, I got a bit confused by `path` thinking it's a file path.

Reply:
Indeed, a few lines more code but it avoids producing something `/path/to/xyz:` (an empty component after `:`).